src/kademlia/dht_tracker.cpp

```diff
@@ -538,29 +530,35 @@ namespace libtorrent { namespace dht {
 #endif
 
 		libtorrent::dht::msg m(m_msg, ep);
-		m_dht.incoming(m);
-#if TORRENT_USE_IPV6
-		m_dht6.incoming(m);
-#endif
+		for (auto& n : m_nodes)
+			n.second.dht.incoming(m);
 		return true;
 	}
 
+	dht_tracker::tracker_node::tracker_node(io_service& ios
+		, aux::session_listen_socket* s, socket_manager* sock
+		, libtorrent::dht_settings const& settings
+		, node_id const& nid
+		, dht_observer* observer, counters& cnt
+		, get_foreign_node_t get_foreign_node
+		, dht_storage_interface& storage)
+		: dht(s, sock, settings, nid, observer, cnt, get_foreign_node, storage)
+		, connection_timer(ios)
+	{}
+
 	std::vector<std::pair<node_id, udp::endpoint>> dht_tracker::live_nodes(node_id const& nid)
 	{
 		std::vector<std::pair<node_id, udp::endpoint>> ret;
 
-		// TODO: figure out a better solution when multi-home is implemented
-		node const* dht = m_dht.nid() == nid ? &m_dht
-#if TORRENT_USE_IPV6
-			: m_dht6.nid() == nid ? &m_dht6 : nullptr;
-#else
-			: nullptr;
-#endif
-
-		if (dht == nullptr) return ret;
+		for (auto& n : m_nodes)
```

Comment:
It ends up being rather terse because I have to spell out the type of m_nodes. I could typedef it but it doesn't seem worth the effort. We can revisit this once we move the codebase to C++14 and can use `auto` parameters in lambdas.